### PR TITLE
v0.7

### PR DESCRIPTION
**Pull request by fi3ework**
# v0.7

Breaking changes:

## 💥 Split check abilities into separate packages

After v0.7, the usage will be like:

package.json
```json
"vite-plugin-checker": "some version", 
"@vite-plugin-checker/eslint": "some version",
"eslint": "some version"
```

vite config
```js
import { defineConfig } from 'vite'
import { checker } from 'vite-plugin-checker'
import { checker as eslint } from '@vite-plugin-checker/eslint'

export default defineConfig({
  plugins: [
    checker(
      [
        eslint({
          lintCommand: 'eslint ./src --ext .ts',
        }),
      ],
      {
        enableBuild: false,
      }
    ),
  ],
})

```

Why?
- Friendly to user, user does not need to install any low-level packages dependented by ESLint for Stylelint.
- vite-plugin-checker's dependencies will be much cleaner. No useless dependencies will be installed.
- For better extensibility of the plugin, the checker count is scalable.

## 💥 Remove default export support 

After v0.7, no default export will be provided from vite-plugin-checker to reduce ESM and CJS potential issues. The named export `checker` will be the only one to use. Also, new added checker packages will not provide default export but also a named export `checker`.

### CODE REVIEW

File: packages/vite-plugin-checker/package.json [hunk 1]
```diff
@@ -42,70 +62,21 @@
...
-    "fast-glob": "^3.2.7",
-    "fs-extra": "^11.1.0",
     "npm-run-path": "^4.0.1",
     "semver": "^7.5.0",
     "strip-ansi": "^6.0.0",
-    "tiny-invariant": "^1.1.0",
+    "tiny-invariant": "^1.3.1",
     "vscode-languageclient": "^7.0.0",
     "vscode-languageserver": "^7.0.0",
     "vscode-languageserver-textdocument": "^1.0.1",
```
Comment (armano2) [inline packages/vite-plugin-checker/package.json:71]: those dependencies should be moved to vls

File: packages/vite-plugin-checker/src/logger.ts [hunk 2]
```diff
@@ -1,15 +1,13 @@
...
-import { createRequire } from 'module'
-import os from 'os'
+import os from 'node:os'
 import strip from 'strip-ansi'
 import * as _vscodeUri from 'vscode-uri'
 
 // hack to compatible with Jiti
 // see details: https://github.com/fi3ework/vite-plugin-checker/issues/197
 // @ts-expect-error
 const URI = _vscodeUri?.default?.URI ?? _vscodeUri.URI
```
Comment (armano2) [inline packages/vite-plugin-checker/src/logger.ts:9]: this "hack" should be moved to places where we import `vscode-uri`, as its no longer used in here we should remove it from this file

File: packages/checker-vue-tsc/src/logger.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    })
+  }
+
+  return {
+    message,
+    conclusion: '',
+    codeFrame,
+    stripedCodeFrame: codeFrame && strip(codeFrame),
+    id: fileName,
+    checker: 'TypeScript',
```
Comment (armano2) [inline packages/checker-vue-tsc/src/logger.ts:56]: we should use `vue-tsc` instead

File: packages/checker-eslint/src/logger.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import type { NormalizedDiagnostic, SourceLocation } from 'vite-plugin-checker/logger'
+import { DiagnosticLevel } from 'vite-plugin-checker/types'
+
+export function normalizeEslintDiagnostic(diagnostic: ESLint.LintResult): NormalizedDiagnostic[] {
+  return diagnostic.messages
+    .map((d) => {
+      let level = DiagnosticLevel.Error
+      switch (d.severity) {
+        case 0: // off, ignore this
+          level = DiagnosticLevel.Error
```
Comment (armano2) [inline packages/checker-eslint/src/logger.ts:12]: this assign is unnecessary

**Comment by fi3ework** [comment]
v0.7 is set to release with an important fix of compatibility with vue-tsc@2. A lot of changes has made since then. It might be possible to release a v0.8 to compatible with ESLint 9. Then will start a 0.10.0 with separate checkers into different packages. I'll start to do this over again.

**Pull request closed without merging**